### ISSUE TITLE
Question how to read the label file?

### ISSUE THREAD

**Issue by craftindo** [facebookresearch/qmnist#1]
Here is my code:

**<this code here is to read the image file>**
import gzip
f = gzip.open('./QMnist/qmnist-test-images-idx3-ubyte.gz','r')
image_size = 28
num_images = 60000
import numpy as np
f.read(16)
#f.read()
#buf = f.read(image_size * image_size * num_images)
buf = f.read()
data = np.frombuffer(buf, dtype=np.uint8).astype(np.float32)
data = data.reshape(num_images, image_size, image_size, 1)
print(data.shape)
import matplotlib.pyplot as plt
plt.imshow(data[1].reshape(28,28))

**<the reader for the label file>**
f = gzip.open('./QMnist/qmnist-train-labels-idx2-int.gz','r')
f.read(12)# skip 12 byte as document says
datalabel = np.frombuffer(buf, dtype=np.uint32).astype(np.int32)
print(datalabel.shape)
datalabel = datalabel.reshape(8,num_images)
print(datalabel.shape)
print(datalabel[0])

**The output**
(480000,)
(8, 60000)
[  83886080   67108864 1778909184 ..., 2769158144          0          0]

Question:
1. the first image is 7(based on my reader code), but i can't relate the class to 7
Please help

**Comment by leonbottou** [comment]
See function `read_idx2_int` in [`qmnist.py`](https://github.com/facebookresearch/qmnist/blob/master/qmnist.py#L187).

For an even more general way to read mnist files, see function `read_sn3_pascalvincent_tensor` in [`torchvision/mnist.py`](https://github.com/pytorch/vision/blob/fa6af6d1e6f050d0a930e6f59894b7dd40aa869c/torchvision/datasets/mnist.py#L443).
